includes/models/newspaper.inc

```diff
@@ -1,42 +1,99 @@
 <?php
-
-
 /**
  * @file
  */
-
 /**
- * This is the actual manifest building function.
+ * Create the requested newspaper manifest. When $selected_collection is set, the manifest of that collection is returned otherwise
+ * the full news paper manifest collection is returned
+ * @param AbstractObject $object
+ * @param null $selected_collection - Return the selected sub collection
+ * @return mixed
  */
-function islandora_iiif_manifest_create_newspaper_manifest(AbstractObject $object) {
+function islandora_iiif_manifest_create_newspaper_manifest(AbstractObject $object, $selected_collection = NULL) {
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
   module_load_include('inc', 'islandora_iiif_manifests', 'includes/utilities');
   module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
   module_load_include('inc', 'islandora_newspaper', 'includes/utilities');
+  module_load_include('inc', 'islandora', 'includes/datastream');
+  
+  if (is_null($selected_collection)) {
+    $manifest = islandora_iiif_manifest_get_newspaper_manifest($object);
+  }
+  else {
+    $manifest = islandora_iiif_manifest_get_newspaper_collection($object, $selected_collection);
+  }
+  
+  return $manifest;
+}
 
-  // Manifest array.
-  $manifest = array();
-
+/**
+ * Create the newspaper manifest with collections and its member within
+ * @param $object
+ * @return mixed
+ */
+function islandora_iiif_manifest_get_newspaper_manifest($object) {
+  // Collections array
+  $manifest['collections'] = array();
   // Get issues.
   $issues = islandora_newspaper_get_issues($object);
-
-  $collection_members_members = array();
-
-  foreach ($issues as $issue) {
-
-    // Load issue.
-    $issue_object = islandora_object_load($issue['pid']);
-
-    $collection_item = islandora_iiif_manifests_get_object_collection($issue_object);
-
-    $collection_members_members[] = $collection_item['object'];
-
+  // Get issues.
+  $group_issues = islandora_newspaper_group_issues($issues);
+  // Collections array
+  $manifest['collections'] = array();
+  foreach ($group_issues as $collection => $member) {
+    
+    $newspaper_member = islandora_iiif_manifests_get_newspaper_member_collection($member);
+    if (!empty($newspaper_member)) {
+      $manifest_uri = urldecode(url(variable_get('islandora_iiif_manifests_domain_uri', $GLOBALS['base_url']) . "/iiif_manifest/{$object->id}/collection/{$collection}", array(
+        'absolute' => TRUE,
+        'language' => (object) array('language' => FALSE),
+      )));
+      if (is_numeric($collection)) {
+        $label = strval($collection);
+      }
+      else {
+        $label = $collection;
+      }
```

Comment:
Isn't it easier to just do:
$label = "$collection";

Reply:
Removed this code and replaced with:

` if (is_numeric($volume_label)) {
        $volume_label = strval($volume_label);
      }
      
      $newspaper_collection = array(
        "@id" => $manifest_uri,
        "@type" => 'sc:Collection',
        "label" => $volume_label,
      );`

Reply:
But if the label is numeric, then you should use quotes around it?

Reply:
@lucasvanschaik You are right! I saw that in other parts of the code I do use quotes.